comp/metadata/inventoryotel/inventoryotelimpl/inventoryotel.go

```diff
@@ -0,0 +1,256 @@
+}
+
+type inventoryotel struct {
+	util.InventoryPayload
+
+	conf      config.Component
+	log       log.Component
+	m         sync.Mutex
+	data      otelMetadata
+	hostname  string
+	authToken authtoken.Component
+	f         *freshConfig
+}
+
+type dependencies struct {
+	fx.In
+
+	Log        log.Component
+	Config     config.Component
+	Serializer serializer.MetricSerializer
+	AuthToken  authtoken.Component
+}
+
+type provides struct {
+	fx.Out
+
+	Comp                 iointerface.Component
+	Provider             runnerimpl.Provider
+	FlareProvider        flaretypes.Provider
+	StatusHeaderProvider status.HeaderInformationProvider
+}
+
+func newInventoryOtelProvider(deps dependencies) provides {
+	hname, _ := hostname.Get(context.Background())
+	i := &inventoryotel{
+		conf:      deps.Config,
+		log:       deps.Log,
+		hostname:  hname,
+		data:      make(otelMetadata),
+		authToken: deps.AuthToken,
+	}
+
+	getter := i.fetchRemoteOtelConfig
+	if i.conf.GetBool("otel.submit_dummy_inventories_metadata") {
+		getter = i.fetchDummyOtelConfig
+	}
+
+	var err error
+	i.f, err = newFreshConfig(deps.Config.GetString("otel.extension_url"), getter)
+	if err != nil {
+		// panic?
+		return provides{}
+	}
+
+	i.InventoryPayload = util.CreateInventoryPayload(deps.Config, deps.Log, deps.Serializer, i.getPayload, "otel.json")
+
+	if i.Enabled {
+
+		// TODO: if there's an update on the OTel side we currently will not be
+		//       notified. Is this a problem? Runtime changes are expected to be
+		//       triggered by FA, so maybe this is OK.
+		//
+		// We want to be notified when the configuration is updated
+		deps.Config.OnUpdate(func(_ string, _, _ any) { i.Refresh() })
+	}
+
+	return provides{
+		Comp:                 i,
+		Provider:             i.MetadataProvider(),
+		FlareProvider:        i.FlareProvider(),
+		StatusHeaderProvider: status.NewHeaderInformationProvider(i),
+	}
+}
+
+func scrub(s string) string {
+	// Errors come from internal use of a Reader interface. Since we are reading from a buffer, no errors
+	// are possible.
+	scrubString, _ := scrubber.ScrubString(s)
+	return scrubString
+}
+
+func (i *inventoryotel) parseResponseFromJSON(body []byte) (otelMetadata, error) {
+	var conf interface{}
+
+	err := json.Unmarshal(body, &conf)
+	if err != nil {
+		i.log.Errorf("Error unmarshaling the response:", err)
+		return nil, err
+	}
+
+	return conf.(otelMetadata), nil
+}
+
+func (i *inventoryotel) fetchRemoteOtelConfig(u *url.URL) (otelMetadata, error) {
+	// Create a Bearer string by appending string access token
+	var bearer = "Bearer " + i.authToken.Get()
+
+	// Create a new request using http
+	req, err := http.NewRequest("GET", u.String(), nil)
+	if err != nil {
+		i.log.Errorf("Error building request: ", err)
+		return nil, err
+	}
+
+	// add authorization header to the req
+	req.Header.Add("Authorization", bearer)
+
+	client := &http.Client{}
```

Comment:
can we use http default client or move this client to the inventoryotel

Reply:
We shouldn't be using the default client, but you're absolutely right, we shouldn't be allocating a client per iteration.